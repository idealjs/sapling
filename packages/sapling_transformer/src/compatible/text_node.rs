use biome_js_factory::make::{
    ident, js_call_expression, js_identifier_expression, js_reference_identifier,
    js_string_literal, js_string_literal_expression,
};
use biome_js_syntax::{
    AnyJsCallArgument, AnyJsExpression, AnyJsLiteralExpression, JsCallExpression,
};

use crate::make_js_call_arguments;

pub fn generate_create_text_node_expr(text: &str) -> JsCallExpression {
    let callee = js_identifier_expression(js_reference_identifier(ident("_$createTextNode")));
    let arg = AnyJsCallArgument::AnyJsExpression(AnyJsExpression::AnyJsLiteralExpression(
        AnyJsLiteralExpression::JsStringLiteralExpression(js_string_literal_expression(
            js_string_literal(text),
        )),
    ));
    js_call_expression(callee.into(), make_js_call_arguments(vec![arg], vec![])).build()
}

#[test]
fn test_generate_create_text_node_expr() {
    let expr = generate_create_text_node_expr("hello");
    // 只做快照测试，确保语法结构正确
    insta::assert_snapshot!(expr.to_string());
}
