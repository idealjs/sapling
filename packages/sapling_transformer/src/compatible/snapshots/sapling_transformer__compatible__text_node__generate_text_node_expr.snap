---
source: packages/sapling_transformer/src/compatible/text_node.rs
expression: expr.to_string()
---
_$createTextNode("hello")
